src/SOS/Strike/eeheap.cpp

```diff
@@ -2207,13 +2207,50 @@ DWORD_PTR JitHeapInfo()
 *    Returns: The size of this heap.                                   *
 *                                                                      *
 \**********************************************************************/
-DWORD_PTR LoaderHeapInfo(CLRDATA_ADDRESS pLoaderHeapAddr, DWORD_PTR *wasted)
+DWORD_PTR LoaderHeapInfo(CLRDATA_ADDRESS pLoaderHeapAddr, LoaderHeapKind kind, DWORD_PTR *wasted)
 {
     g_trav_totalSize = 0;
     g_trav_wastedSize = 0;
 
     if (pLoaderHeapAddr)
-        g_sos->TraverseLoaderHeap(pLoaderHeapAddr, LoaderHeapTraverse);
+    {
+        ReleaseHolder<ISOSDacInterface13> sos13;
+        if (SUCCEEDED(g_sos->QueryInterface(__uuidof(ISOSDacInterface13), &sos13)))
+        {
+            // If we have ISOSDacInterface13, simply use the working TraverseLoaderHeap.
+
+            sos13->TraverseLoaderHeap(pLoaderHeapAddr, kind, LoaderHeapTraverse);
+        }
+        else
+        {
+            // If we don't have ISOSDacInterface13, any ExplicitControl loader heap needs to have
+            // a "fake" vtable pointer.  To do this we simply subtract off a pointer...since the vtable
+            // is not used/verified this works fine.
+            // On .Net 7, the logic was inverted for just that release.  For that runtime, we add a
+            // pointer to LoaderHeap pointers to remove the vtable.
+
+            VS_FIXEDFILEINFO verInfo = { 0 };
+
+            bool isNetCore7 = g_pRuntime->GetRuntimeConfiguration() != IRuntime::RuntimeConfiguration::WindowsDesktop;
+            isNetCore7 &= SUCCEEDED(g_pRuntime->GetEEVersion(&verInfo, nullptr, 0));
```

Comment:
There is a helper function in util.cpp `bool IsRuntimeVersion(DWORD major)` that can do all of this. I'm not sure you need to check for desktop Framework either because it will never have a major version of 7.

Reply:
Ah, thanks!  I missed that.